src/coordinator/ts/series.go

```diff
@@ -49,3 +54,64 @@ func (b *Series) Len() int { return b.vals.Len() }
 
 // Values returns the underlying values interface
 func (b *Series) Values() Values { return b.vals }
+
+// Align adjusts the datapoints to start, end and a fixed interval
+func (b *Series) Align(start, end time.Time, interval time.Duration) (*Series, error) {
+	fixedVals, err := alignValues(b.Values(), start, end, interval)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSeries(b.name, fixedVals, b.Tags), nil
+}
+
+func alignValues(values Values, start, end time.Time, interval time.Duration) (FixedResolutionMutableValues, error) {
```

Comment:
nit: make this a receiver function on *Series and use b.Values()

Reply:
i kept it separate since it's easier to test this way (Although i need to write more tests :D). This also makes it clearer that the underlying series isn't being mutated.

Reply:
Might be better to call it something like `generateAlignedValues` or something then? `alignValues` implies mutation